netaddr_test.go

```diff
@@ -1323,3 +1323,33 @@ func TestAs4(t *testing.T) {
 		}
 	}
 }
+
+func TestIPPrefixOverlaps(t *testing.T) {
+	pfx := mustIPPrefix
+	tests := []struct {
+		a, b IPPrefix
+		want bool
+	}{
+		{IPPrefix{}, pfx("1.2.0.0/16"), false},  // first zero
```

Comment:
It seems to me like you'd want 0/0 (or 0.0.0.0/0 if it doesn't allow omitting the unnecessary IP parts) to return true for anything, e.g. in cases where you're checking if something matches a default route.

Reply:
Note that this is a zero value, not a value representing a `/0`.

The contract as documented is you must provide two valid values to compare. Otherwise we just return false. (The alternative perhaps might be to panic, but that seems aggressive.)

Reply:
Oh, if 0.0.0.0/0 and IsZero() are distinct, then that's great. Maybe just add a test verifying that it does indeed work, since that seems like the sort of subtle thing that could break by accident.